### PR TITLE
Additional updates related to Swift 4.2

### PR DESCRIPTION
**Pull request by norio-nomura**

### CODE REVIEW

File: Tests/SourceKittenFrameworkTests/DocInfoTests.swift [hunk 1]
```diff
@@ -22,9 +22,9 @@ class DocInfoTests: XCTestCase {
     }
 
     func testModuleInfoRequest() throws {
-#if swift(>=4.1.50)
+#if swift(>=4.2)
```
Comment (jpsim) [inline Tests/SourceKittenFrameworkTests/DocInfoTests.swift]: Why this change? Same compiler version, no?

File: Source/SourceKittenFramework/LinuxCompatibility.swift [hunk 2]
```diff
@@ -10,50 +10,30 @@ import Foundation
 
 extension Array {
     public func bridge() -> NSArray {
```
Comment (jpsim) [inline Source/SourceKittenFramework/LinuxCompatibility.swift]: Should we remove all these bridge functions?

File: Tests/SourceKittenFrameworkTests/DocInfoTests.swift (hunk 1 above)
Comment (norio-nomura) [reply Tests/SourceKittenFrameworkTests/DocInfoTests.swift]: By this PR, SourceKitten will be built with `-swift-version 4.2` swift compiler option.
Under that condition, it will never be `4.1.50`, so I thought that this should be changed.
🤔 
Or should we switch to [compiler version directive](https://github.com/apple/swift-evolution/blob/master/proposals/0212-compiler-version-directive.md) on occasion?

File: Source/SourceKittenFramework/LinuxCompatibility.swift (hunk 2 above)
Comment (norio-nomura) [reply Source/SourceKittenFramework/LinuxCompatibility.swift]: It may be so, but I think it would be better to have another PR.

Comment (jpsim) [reply Source/SourceKittenFramework/LinuxCompatibility.swift]: Agreed

File: Tests/SourceKittenFrameworkTests/DocInfoTests.swift (hunk 1 above)
Comment (jpsim) [reply Tests/SourceKittenFrameworkTests/DocInfoTests.swift]: In some cases maybe we should but that’s unnecessary right now.
